aiida/tools/importexport/dbexport/__init__.py

```diff
@@ -104,6 +104,10 @@ def export(
         Support for the parameter `silent` will be removed in `v2.0.0`.
         Please set the log level and progress bar implementation independently.
 
+    .. deprecated:: 1.5.0
+        Support for the parameter `use_compression` will be removed in `v2.0.0`.
+        Please use `writer_init={'use_compression': True}`.
```

Comment:
I didn't quite follow the logic of which options go into the `writer_init` and which don't - `writer_init` in my view is not very specific.
E.g. from a user perspective,`file_format`, `overwrite`, etc. also specify aspects of how the file should be written.

I guess you introduced the `writer_init` in order to avoid having yet another parameter for activating the low-memory option.
I think it's better to add it (or - what I would prefer - to stick to the strings for `file_format` and then do the translation inside the `export` function).

Reply:
Ah hm.. I see that 

> or - what I would prefer - to stick to the strings for `file_format` and then do the translation into classes inside the `export` function

Ah hm, I see that `file_format` here is already a translated version of the string passed on the CLI, so that suggestion doesn't work.
One could make it a separate class - or add the additional parameter.

Reply:
> I didn't quite follow the logic of which options go into the writer_init and which don't

simply options that are writer specific. It is misleading to have `use_compression` as a core option, because it does not apply to tar files.

> I guess you introduced the writer_init in order to avoid having yet another parameter for activating the low-memory option.

no `writer_init` was already in the code way before this

> One could make a separate class for the writer with zipinfo caching 

no I disagree, you should not have to make a separate class for every possible option, this is what `writer_init` is for.

Reply:
essentially `file_format` could now maybe be more accurately described as `writer_cls`, which either points to the "entry-point name" or actual class

Reply:
The main issue I have here is that this additional nesting is rather non-obvious  for a user who shouldn't have to care about the internal implementation - while the other parameter names are somewhat self-explanatory,`writer_init` is rather opaque.

If we could find a way to make things more obvious that would be very useful - `export` is *the* function people are going to use to create export files from scripts, so we should try to make it easy to use.

Reply:
>  a user who shouldn't have to care about the internal implementation

these are only to set fine-grained control of the writer though, above the defaults, which should be fine for most users.
Its more for "power" users that want to create their own writer classes.
I would use **kwargs, but that is already taken by the graph traversal rules lol

Reply:
removed suggestion to use writer init for use_compression (as discussed) in 98f0b9d